syz-verifier/verf/verifier.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2021 syzkaller projelct authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+package verf
+
+import (
+	"log"
+	"math/rand"
+
+	"github.com/google/syzkaller/pkg/ipc"
+	"github.com/google/syzkaller/pkg/mgrconfig"
+	"github.com/google/syzkaller/pkg/report"
+	"github.com/google/syzkaller/prog"
+	"github.com/google/syzkaller/vm"
+)
+
+// Verifier TODO
+type Verifier struct {
+	Kis         []*KernelInfo
+	Crashdir    string
+	VmStop      chan bool
+	Target      *prog.Target
+	RunnerBin   string
+	ExecutorBin string
+	ChoiceTable *prog.ChoiceTable
+	Rnd         *rand.Rand
+	Addr        string
+}
+
+// KernelInfo contains kernel-specific information, necessary to spawn virtual  // machines and report crashes.
+type KernelInfo struct {
+	Cfg      *mgrconfig.Config
+	VmPool   *vm.Pool
+	Reporter report.Reporter
+}
+
+// Result stores the execution results of a program.
+type Result struct {
+	// RnName is the name of the Runner that executed the program
+	RnName string
+	// ProgIdx is the index of the program that produced results
+	ProgIdx int
+	// Hanged is set to true when a program was killed due to hanging.
+	Hanged bool
+	// Info contains information about the execution of each system call in the generated programs.
+	Info ipc.ProgInfo
+}
+
+// Verify checks whether the Results of the same program, executed by           // several Runners are the same.
+// TODO: Decide what to do if Hanged is set to true.
+// TODO: Extend the verification
+// TODO: Add a storage mechanism of interesting results.
+func (vrf *Verifier) Verify(res []*Result, sp []byte) {
+	log.Printf("%d %d", len(res), res[0].ProgIdx)
+	p, err := vrf.Target.Deserialize(sp, prog.NonStrict)
```

Comment:
We generate the program in normal, deserialized representation. We don't need to serialize and deserialize it back. Just pass *Prog around directly.